cmd/dist-pull/dist-pull

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+# This is a quick and dirty hack until containerd supports a higher
+# level API via gRPC than the command line dist tool.
+set -e
+
+if [ $# -ne 3 ] ; then
+    echo "usage: <store> <remote> <object>" >&2
+    exit 1
+fi
+
+store=$1
+remote=$2
+object=$3
+
+manifest=$(dist fetch $remote $object mediatype:application/vnd.docker.distribution.manifest.v2+json)
+
+env="env remote=$remote store=$store"
+mkdir -p $store/manifests
+
+mpath="$store/manifests/$remote:$object"
+
+mkdir -p $(dirname "$mpath")
+echo "$manifest" | sudo dd of=$mpath
+
+schema=$(echo "$manifest" | jq .schemaVersion)
+case $schema in
+    1)
+	echo "$manifest" | \
+	    jq -r '.fsLayers[] | "dist fetch $remote "+.blobSum + "| dist ingest --path=$store --expected-digest "+.blobSum' | \
+	    sort -u | $env xargs -I{} -P10 -n1 sh -c "{}"
+	    ;;
+    2)
+	echo "$manifest" | \
+	    jq -r '.layers[] | "dist fetch $remote "+.digest + " mediatype:"+.mediaType+"| dist ingest --path=$store --expected-digest "+.digest' | \
+	    sort -u | $env xargs -I{} -P10 -n1 sh -c "{}"
+
+	echo "$manifest" | \
+	    jq -r '.config | "dist fetch $remote "+.digest + " mediatype:"+.mediaType+"| dist ingest --path=$store --expected-digest "+.digest' | \
+	    sort -u | $env xargs -I{} -P10 -n1 sh -c "{}"
+	;;
+    *)
+	echo "Unknown schema $schema" >&2
+	exit 1
+	;;
+esac
+
+exit 0
```

Comment:
You might be able to hack up a rootfs by running each layer through `dist apply`.